src/components/List/Cells/ApplicantActionCell.tsx

```diff
@@ -4,17 +4,22 @@ import { ApplicationStatus } from '../../../utils/generated/graphql'
 import { CellProps } from '../../../utils/types'
 import { useLanguageProvider } from '../../../contexts/Localisation'
 import { Icon } from 'semantic-ui-react'
+import { useRouter } from '../../../utils/hooks/useRouter'
 
 const ApplicantActionCell: React.FC<CellProps> = ({ application: { status, serial } }) => {
   const { t } = useLanguageProvider()
+  const { location } = useRouter()
   let action = ''
 
   if (status === ApplicationStatus.ChangesRequired) action = t('ACTION_UPDATE')
   if (status === ApplicationStatus.Draft) action = t('ACTION_CONTINUE')
 
   if (!action)
     return (
-      <Link className="user-action" to={`/application/${serial}`}>
+      <Link
+        className="user-action"
+        to={{ pathname: `/application/${serial}`, state: { prevQuery: location.search } }}
+      >
```

Comment:
So the basic idea is that we can pass additional state through a link using this syntax. In this case we're passing the filter string so we can use it to generate the "Back" link in the linked component. :)